crates/bevy_ecs/src/lib.rs

```diff
@@ -1262,6 +1262,37 @@ mod tests {
         assert_eq!(world.resource::<A>().0, 1);
     }
 
+    #[test]
+    fn non_send_resource_scope() {
+        let mut world = World::default();
+        world.insert_non_send_resource(A(0));
```

Comment:
I think we should probably use a different resource type that is deliberately not Send.